### PR TITLE
about:version:help clean up add Bitcoin branding

### PR DESCRIPTION
**Pull request by RandyMcMillan**
Before:

Screen Shot 2020-11-29 at 3 26 26 AM

After:

> Updated default presentations:
> 
> Screen Shot 2020-12-04 at 3 19 07 PM
> 
> Screen Shot 2020-12-04 at 3 18 56 PM

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -591,23 +591,28 @@ void SetupServerArgs(NodeContext& node)
...
+{
+    return FormatVersion();
+}
+
+std::string CopyrightInfo()
+{
+    return CopyrightHolders(strprintf(_("Copyright (C) %i-%i").translated, 2009, COPYRIGHT_YEAR) + " ");
+}
+
+std::string AboutMessageInfo()
```
Comment (RandyMcMillan) [inline src/init.cpp:604]: The AboutMessageInfo() is simple and can be modified more easily in the future.

File: src/init.h [hunk 2]
```diff
@@ -68,7 +68,16 @@ bool AppInitMain(const util::Ref& context, NodeContext& node, interfaces::BlockA
  */
 void SetupServerArgs(NodeContext& node);
 
-/** Returns licensing information (for -version) */
+/** Returns versionInfo text for utilitydialogue  */
```
Comment (RandyMcMillan) [inline src/init.h:71]: utilitydialogue is mentoned in the comments for better documentation.

File: src/qt/bitcoin.qrc [hunk 3]
```diff
@@ -44,6 +44,7 @@
         <file alias="hd_disabled">res/icons/hd_disabled.png</file>
         <file alias="network_disabled">res/icons/network_disabled.png</file>
         <file alias="proxy">res/icons/proxy.png</file>
+        <file alias="bitcoin_core">res/icons/bitcoin_core.png</file>
```
Comment (RandyMcMillan) [inline src/qt/bitcoin.qrc:47]: Added the png art that is used in the header of the about view.

File: src/qt/forms/helpmessagedialog.ui [hunk 4]
```diff
@@ -6,10 +6,13 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>780</width>
-    <height>400</height>
+    <width>602</width>
```
Comment (RandyMcMillan) [inline src/qt/forms/helpmessagedialog.ui:9]: The new configuration makes a better use of space - therefore the window doesn't need to open as big.

File: src/qt/utilitydialog.cpp [hunk 5]
```diff
@@ -41,21 +41,37 @@ HelpMessageDialog::HelpMessageDialog(QWidget *parent, bool about) :
         setWindowTitle(tr("About %1").arg(PACKAGE_NAME));
 
         std::string licenseInfo = LicenseInfo();
-        /// HTML-format the license message from the core
+        std::string copyrightInfo = CopyrightInfo();
```
Comment (RandyMcMillan) [inline src/qt/utilitydialog.cpp:44]: The new implementation separates these strings.

**Comment by hebasto** [comment]
NACK on changes of `-version` command-line argument output layout.

**Comment by jarolrod** [comment]
not-aligned
These two sections should be aligned

**Comment by RandyMcMillan** [comment]
Updated Screenshots:


Screen Shot 2020-11-30 at 2 37 28 PM
Screen Shot 2020-11-30 at 2 37 39 PM

**Comment by RandyMcMillan** [comment]
@hebasto 

I ensured that the output is uniform for -version

thanks for the feedback! :) 

Screen Shot 2020-11-30 at 11 45 04 PM

**Comment by jonasschnelli** [comment]
I like the new style.
Though there is still a debate on whether the logo should be orange or black.
https://github.com/bitcoin-core/gui/issues/89

Concept ACK if we keep the color as is (we can always change it later to black if people want this).

**Comment by RandyMcMillan** [comment]
@jonasschnelli - Since it is the about page - I thought using the web page logo might be useful. As it is now, the aboutMessage body is set up for further development - I was thinking of adding a menu bar that would replace the hyperlinks (It will look better). 

A note on color usage and logo specification - keep in mind that most companies/entities create specifications for full color - gray scale and black/white - for different purposes such as web design, print, what ever - As far as I'm concerned - there really isn't a debate - because we should have a specification and media resource files that fit print/web/etc use cases. 

Here is an example of a logo specification

https://ori.hhs.gov/logo-specification-guide

Screen Shot 2020-12-02 at 11 32 02 PM

**Comment by RandyMcMillan** [comment]
@jonasschnelli - Since it is the about page - I thought using the web page logo might be useful. As it is now the aboutMessage body is set up for further development - I was thinking of adding a menu bar that would replace the hyperlinks (It will look better). 

A note on color usage - keep. in mind that most companies create specifications for full color - gray scale and black - for different purposes ass web design, print, what ever - As far as I'm concerned - there really isn't a debate - because we should have a specification and media resource files that fit print/web/etc use cases.

**Comment by Bosch-0** [comment]
Concept ACK

The updated screenshots you posted above, the text should align with the logo such as below. 

image

**Comment by bitcoinheiro** [comment]
I would change that logo. This is Bitcoin, Core is secondary. Only forks want to reinforce the Core over Bitcoin.

- Orange logo
- Bitcoin bold
- Core thin

**Comment by Bosch-0** [comment]
> I would change that logo. This is Bitcoin, Core is secondary. Only forks want to reinforce the Core over Bitcoin.

I disagree with this view. The GUI this repo is for is a product developed by Core that uses the Bitcoin protocol. Other implementations may one day exist that work on Bitcoin that aren't Core.

**Comment by bitcoinheiro** [comment]
What do we want to communicate?
Do we want to have more force on the name Bitcoin or Core?

We cannot be always thinking about other people concerns, the software is what it is and other implementations can exist - and they do exist - without the need to reinforce this concept on the logo and be "apologetic".

I believe users of the Core implementation should feel they are running BITCOIN when running this "product" and not "Core".
When I see the black logo it makes me feel sad, it confuses me instead of reinforcing Bitcoin. Feels like you're putting Bitcoin "in the closet" and we should be ashamed to be Bitcoin instead of proud of it.

**Comment by Bosch-0** [comment]
>  Do we want to have more force on the name Bitcoin or Core?

One or the other at this stage - currently both are used and its confusing. 

> I believe users of the Core implementation should feel they are running BITCOIN

You can still achieve this whilst using the black logo - do people think they are running something else when using products like electrum or bluewallet? I'd say no. This is the designers job to communicate this effectively and use the black and orange logos in their appropriate locations. 

Do most people see Bitcoin Core = Bitcoin?

**Comment by bitcoinheiro** [comment]
> Do most people see Bitcoin Core = Bitcoin?
I believe people do and you're exactly trying to use the design to communicate otherwise.
I don't think people have to be communicated otherwise.

Why Bitcoin so thin and Core so bold? Isn't that trying to communicate Core over Bitcoin?
This project is not Electrum or BlueWallet, it's Bitcoin.

image

**Comment by Bosch-0** [comment]
I've already explained my rationale many times above.

I'm still of the thought that:

Bitcoin Core is a group of developers who work on Bitcoin.

Bitcoin is a network that everyone is apart of. 

Electrum and BlueWallet are part of the Bitcoin ecosystem as is the GUI developed by the Core developers.

**Comment by jonatack** [comment]
I agree with not emphasizing "Core". Let's not institutionalize that emphasis further. Please keep them equal or put more weight on "Bitcoin".

One of my pet peeves is people writing "Core" instead of "Bitcoin Core" ("Core devs", "running Core", and so on).

See the [Bitcoin Optech style guide](https://github.com/bitcoinops/bitcoinops.github.io/blob/master/STYLE.md):

```
Forbidden abbreviations:

Core (use Bitcoin Core)
```

100% agree with our Bitcoin Optech colleagues about this.

**Comment by jonasschnelli** [comment]
Don't get me wrong. I originally designed the black logo and I like it.
Though @gmaxwell made a valid comment [here](https://github.com/bitcoin-core/gui/issues/89#issuecomment-705384143).

Changing the color might have greater impact than originally anticipated.

**Comment by jonatack** [comment]
Concept ACK, sorry about not reviewing the PR itself, in general this looks very nice @RandyMcMillan and is a good improvement.

**Review by promag — COMMENTED** [review]
I'd be happy to ACK everything but the logo change, maybe you want to split the PR?

**Comment by RandyMcMillan** [comment]
Thank you for the feed back!

**Comment by RandyMcMillan** [comment]
I agree with the statement 

`
The Bitcoin Core Developers
`

 is a group and this is why I've maintained that "Developers" should be capitalized in the branding.

Further more...

From a copyright perspective - THE BITCOIN CORE DEVELOPERS should be in all caps - but this isn't mandated in any legalese that I have read...

But that is outside the scope of this PR. :)

**Comment by RandyMcMillan** [comment]
@promag @jonatack 

This approach is better. I am only using text styling to create "Bitcoin Core" and reusing the existing
bitcoin logo - this doesn't add any weight to the repo.

---

 I've included "Core" to maintain existing branding continuity - but this can easily be removed if/when neccesary.

Screen Shot 2020-12-03 at 7 18 22 PM

---

**Comment by RandyMcMillan** [comment]
I didn't see the link - agree.

https://github.com/bitcoin-core/gui/issues/89#issuecomment-705384143

> The logo should remain orange. This change would inadvertantly play into efforts to create a false equivalence between Bitcoin and various fraudulent clones.
> 
> We now know through leaked emails that even the renaming of Bitcoin Core to that name was dishonestly driven by malicious intent-- the parties that pushed for the name change did so because they were intending to attempt to take over Bitcoin from the community and move it under their control, and they wanted to "other" the community project and disassociate it with Bitcoin. Going along with it was an error. It's water under the bridge now but there is no reason to continue being victimized by past wrongs and compounding a past error.

**Review by jarolrod — COMMENTED** [review]
Tested ACK 5210082
NACK on the current logo text, it looks really bad IMO. 
logo


Also the `about page` and `command-line options page` don't need to be the same size. There's a lot of unused space in the `about page` because they're set to be the same size.

**Comment by RandyMcMillan** [comment]
@jarolrod - please add some context to the pic you posted.
OS - Screen resolution - dark theme? etc...
Thanks.

**Comment by jarolrod** [comment]
> @jarolrod - please add some context to the pic you posted.
> OS - Screen resolution - dark theme? etc...
> Thanks.

OS: Arch Linux, DE: KDE Plasma 5.19.5, Res:3440x1440
The picture is pixelated because its cropped. Below is the full photo. I was trying to point out that everything looks off. The Bitcoin logo is in a weird position in relation to the logo text and the seperator bar. The logo text appears to have weird spacing. I'm not a designer, but this looks off to me.
comand-line-options-page

**Comment by RandyMcMillan** [comment]
@jarolrod 

It is important to consider the way the text is rendered - keep in mind that we select a font family.
Due to the issues with licensing we don't embed a font in the app - so we select a font family - then your local system selects that actual font that is used. There will be variations.


Screen Shot 2020-12-03 at 9 49 19 PM

**Comment by RandyMcMillan** [comment]
Removed excessive space in the aboutMessage body.
These are the default dimensions when they first appear.

Screen Shot 2020-12-03 at 10 51 00 PM

File: src/qt/bitcoingui.cpp [hunk 6]
```diff
@@ -846,6 +846,7 @@ void BitcoinGUI::showDebugWindowActivateConsole()
 void BitcoinGUI::showHelpMessageClicked()
 {
     helpMessageDialog->show();
+    helpMessageDialog->raise();//bring to top if obscured
```
Comment (RandyMcMillan) [inline src/qt/bitcoingui.cpp:849]: If the helpMessageDialog has already been presented - this change brings it to the front if the user re-clicks on the help menu item - otherwise it remains obscured by any other windows.

File: src/qt/utilitydialog.cpp [hunk 7]
```diff
@@ -33,29 +33,46 @@ HelpMessageDialog::HelpMessageDialog(QWidget *parent, bool about) :
     ui(new Ui::HelpMessageDialog)
 {
     ui->setupUi(this);
-
+    this->resize(500,0);//force view height to stretch
```
Comment (RandyMcMillan) [inline src/qt/utilitydialog.cpp:36]: The Ui::HelpMessageDialog is shared with the about view.
These changes force the view to stretch to an appropriate height -  this eliminates excessive height dimensions - the user resizes based upon personal preference.

File: src/qt/utilitydialog.cpp [hunk 8]
```diff
@@ -33,29 +33,46 @@ HelpMessageDialog::HelpMessageDialog(QWidget *parent, bool about) :
...
+    ui->aboutMessage->resize(0,0);//force label to stretch
```
Comment (RandyMcMillan) [inline src/qt/utilitydialog.cpp:37]: https://github.com/bitcoin-core/gui/pull/140/commits/e0ff8e263dc96199385e31e481e64ad3a242fcd9#r536333863

File: src/qt/utilitydialog.cpp [hunk 9]
```diff
@@ -33,29 +33,46 @@ HelpMessageDialog::HelpMessageDialog(QWidget *parent, bool about) :
...
+        ui->scrollArea->setVerticalScrollBarPolicy(Qt::ScrollBarAsNeeded);
+        ui->copyrightInfo->setText(copyrightInfoHTML); //about bottom left corner
+        ui->aboutMessage->setText(aboutMessageInfoHTML); //aboutMessage bodyy
+        ui->licenseInfo->setText(licenseInfoHTML);
+        ui->versionInfo->setText(versionInfoHTML);
+
+
+    if (about)
+    {
+        setWindowTitle(tr("About %1").arg(PACKAGE_NAME));
```
Comment (RandyMcMillan) [inline src/qt/utilitydialog.cpp:73]: Over riding PACKAGE_NAME is outside the scope of this PR.

This PR is implementing changes based on broader community discussions and sentiment.

REF: https://github.com/bitcoin-core/gui/issues/89#issuecomment-705384143

REF: https://github.com/bitcoin-core/gui/pull/140#issuecomment-737773556

File: src/bitcoind.cpp [hunk 10]
```diff
@@ -55,10 +55,10 @@ static bool AppInit(int argc, char* argv[])
 
     // Process help and version before taking care about datadir
     if (HelpRequested(args) || args.IsArgSet("-version")) {
-        std::string strUsage = PACKAGE_NAME " version " + FormatFullVersion() + "\n";
+        std::string strUsage = PACKAGE_NAME " version " + FormatFullVersion() + "\n";//from init.cpp
 
         if (args.IsArgSet("-version")) {
-            strUsage += FormatParagraph(LicenseInfo()) + "\n";
+            strUsage += FormatParagraph(LicenseInfo()) + "\n" + CopyrightInfo() + "\n";//from init.cpp
```
Comment (RandyMcMillan) [inline src/bitcoind.cpp:61]: bitcoind.cpp uses the same strings that help and about use.
LicenseInfo() and CopyrightInfo() are now refactored for reusability.

File: src/clientversion.cpp [hunk 11]
```diff
@@ -47,6 +47,11 @@ static std::string FormatVersion(int nVersion)
     return strprintf("%d.%d.%d", nVersion / 10000, (nVersion / 100) % 100, nVersion % 100);
 }
 
+std::string FormatVersion()
```
Comment (RandyMcMillan) [inline src/clientversion.cpp:50]: Returns a terse string for reusability.
REF: https://github.com/bitcoin-core/gui/pull/140/commits/e0ff8e263dc96199385e31e481e64ad3a242fcd9#r536344384

File: src/init.cpp [hunk 12]
```diff
@@ -591,23 +591,28 @@ void SetupServerArgs(NodeContext& node)
     argsman.AddHiddenArgs(hidden_args);
 }
 
+std::string VersionInfo()
```
Comment (RandyMcMillan) [inline src/init.cpp:594]: REF: https://github.com/bitcoin-core/gui/pull/140/commits/e0ff8e263dc96199385e31e481e64ad3a242fcd9#r536344384

REF: https://github.com/bitcoin-core/gui/pull/140/commits/e0ff8e263dc96199385e31e481e64ad3a242fcd9#r536346038

File: src/qt/bitcoin.qrc (hunk 3 above)
Comment (RandyMcMillan) [reply src/qt/bitcoin.qrc]: Reuse of bitcoin.png is a better approach. Thanks for the feed back!

File: src/qt/forms/helpmessagedialog.ui (hunk 4 above)
Comment (RandyMcMillan) [reply src/qt/forms/helpmessagedialog.ui]: The view's dimensions are now dynamic and resize based on help/about context.

**Comment by RandyMcMillan** [comment]
Updated default presentations:

Screen Shot 2020-12-04 at 3 19 07 PM

Screen Shot 2020-12-04 at 3 18 56 PM

File: src/qt/bitcoin.qrc [hunk 13]
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE RCC><RCC version="1.0">
+<RCC>
```
Comment (jonasschnelli) [inline src/qt/bitcoin.qrc:1]: is this change necessary

File: src/qt/utilitydialog.cpp [hunk 14]
```diff
@@ -33,29 +33,46 @@ HelpMessageDialog::HelpMessageDialog(QWidget *parent, bool about) :
...
+        std::string copyrightInfo = CopyrightInfo();
+        // HTML-format aboutMessage elements
+        QString versionInfoHTML = QString::fromStdString(FormatVersion());
+        QString copyrightInfoHTML = QString::fromStdString(CopyrightInfo());
+        QString aboutMessageInfoHTML = QString::fromStdString(AboutMessageInfo());
         QString licenseInfoHTML = QString::fromStdString(LicenseInfo());
         // Make URLs clickable
         QRegExp uri("<(.*)>", Qt::CaseSensitive, QRegExp::RegExp2);
         uri.setMinimal(true); // use non-greedy matching
+        versionInfoHTML.replace(uri, "<a href=\"\\1\">\\1</a>");
```
Comment (jonasschnelli) [inline src/qt/utilitydialog.cpp:50]: why does it create a link here?

Comment (RandyMcMillan) [reply src/qt/utilitydialog.cpp:50]: You are correct! This string no longer needs this transform.
Awesome!

File: src/qt/bitcoin.qrc (hunk 13 above)
Comment (RandyMcMillan) [reply src/qt/bitcoin.qrc:1]: My concern with this kind of Qt Creator generated diff junk is that it forces Gui developers to
manually edit files to eliminate this stuff. This - I suspect - has contributed to bringing Gui dev to a crawl.
Technically this is valid Qt Creator generated code. I would petition members to reevaluate the policy concerning 
Qt Creator diff junk.

--- 

Please consider my comments here:

In this situation Qt Creator generates elements that it is expecting:
When I removed them it logged this error.

https://github.com/bitcoin-core/gui/pull/135#discussion_r535574885

---

PS I am beginning to suspect that when elements are missing that are expected by Qt - it triggers some kind of error handling in the Qt framework.
This is probably wasting cycles and is slowing down the over all performance of the Gui itself.
Only a theory at this point - I haven't been had time to research it further.

**Comment by RandyMcMillan** [comment]
The code that is blocking this is NOT correct and should be considered a bug.
It omits the if logic and outputs bitcoind -h instead of the expected (-v) version string.

**Comment by RandyMcMillan** [comment]
@practicalswift 

The conflicting code is NOT correct.
This change should have never been ACKd

It removed the if logic necessary to select help (-h) versus version (-v) output.

https://github.com/bitcoin-core/gui/commit/6690adba08006739da0060eb4937126bdfa1181a#diff-214c41e31f440626f2b20c3858589d9270c843e254f38f2f97f5098ca83ef886R60

**Comment by RandyMcMillan** [comment]
Posted PR in main repo:

https://github.com/bitcoin/bitcoin/pull/20597

File: src/bitcoind.cpp [hunk 15]
```diff
@@ -55,13 +55,13 @@ static bool AppInit(int argc, char* argv[])
...
-            strUsage += FormatParagraph(LicenseInfo()) + "\n";
-        } else {
-            strUsage += "\nUsage:  bitcoind [options]                     Start " PACKAGE_NAME "\n";
-            strUsage += "\n" + args.GetHelpMessage();
+        if (!args.IsArgSet("-version")) {
+            strUsage += FormatParagraph(LicenseInfo()) + "\n"
+                "\nUsage:  bitcoind [options]                     Start " PACKAGE_NAME "\n"
+                "\n";
+            strUsage += args.GetHelpMessage();
         }
```
Comment (RandyMcMillan) [inline src/bitcoind.cpp:65]: Resolved conflict.

Comment (RandyMcMillan) [reply src/bitcoind.cpp:65]: Command Line outputs are the same now.

Screen Shot 2020-12-07 at 10 55 22 PM

**Comment by RandyMcMillan** [comment]
When I started this - I didn't realize how badly the Gui needs to be abstracted away from the command line.
I believe this PR is a good step toward this ideal.

**Comment by Rspigler** [comment]
Whether I agree with the change or not, I don't see how it can be appropriate to slip in a name change.  Either the PR title should be changed, or a new PR created IMO.  It also doesn't make sense to me to have the name be Bitcoin Core on the loading screen, as well as `Help`->`About Bitcoin Core` yet then the about page say Bitcoin (without Core). It's just more inconsistency.

We also have other issues underway on branding discussing this issue that aren't cleared yet (#89, #100).  There's a lot of inconsistency.  Bitcoin.org uses orange; bitcoincore.org uses both; bitcoin repo uses black; bitcoin-core repo uses orange.  And now we're making additional naming changes before any decisions have been made.  Seems very messy.

**Comment by RandyMcMillan** [comment]
@Rspigler - I agree it is a pretty serious issue...

There is a lot to do - and lot of decisions to be made.
This emphasizes the importance of a formal specification for branding/logo usage.

And yes...the scope of this PR has changed.
 
More importantly is - Working on the Gui has led me to believe there is a relatively easy work around to abstracting the Gui variables (PROJECT_NAME, etc..) from the build environment variables so Gui dev can progress and the plumbing can be refactored independently...

So once I know which direction to go concerning naming/branding/logo/etc...
How to proceed should be clearer...
